Sources/POSIX/stat.swift

```diff
@@ -10,14 +10,44 @@
 
 import libc
 
-/// Get file status.
-//
-// FIXME: We should probably return our own wrapper type to insulate clients
-// from platform dependencies.
-public func stat(_ path: String) throws -> libc.stat {
-    var buf = libc.stat()
-    guard libc.stat(path, &buf) == 0 else {
-        throw SystemError.dirfd(errno, path)
-    }
-    return buf
+/// Extensions to the libc `stat` structure to interpret the contents in more readable ways.
+extension libc.stat {
+
+     /// File system entity kind.
+     public enum Kind {
+         case file, directory, symlink, fifo, blockdev, chardev, socket, unknown
+
+         private init?(mode: mode_t) {
+             switch mode {
+                 case S_IFREG:  self = .file
+                 case S_IFDIR:  self = .directory
+                 case S_IFLNK:  self = .symlink
+                 case S_IFBLK:  self = .blockdev
+                 case S_IFCHR:  self = .chardev
+                 case S_IFSOCK: self = .socket
+             default:
+                 return nil
```

Comment:
I left `unknown` in the enum case by mistake. I like to indicate unknown by optionals more than a case for them. kind might exist but it won't be our wrapper `Kind`. What do you think?